### PR TITLE
Move e2e test image from centos to alpine

### PR DESCRIPTION
**Pull request by cclerget**
## Description of the Pull Request (PR):

Move e2e test image from centos to alpine to speed up test bootstrap

#### Before submitting a PR, make sure you have done the following:

- Read the [Guidelines for Contributing](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md), and this PR conforms to the stated requirements.
- Added changes to the [CHANGELOG](https://github.com/sylabs/singularity/blob/master/CHANGELOG.md) if necessary according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md)
- Added tests to validate this PR and tested this PR locally with a `make testall`
- Based this PR against the appropriate branch according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md)
- Added myself as a contributor to the [Contributors File](https://github.com/sylabs/singularity/blob/master/CONTRIBUTORS.md)

Attn: @singularity-maintainers

### CODE REVIEW

**Review by mem — CHANGES_REQUESTED** [review]
I don't get it. Rebuild time for e2e is minimal.

What problem are you actually solving by doing this?

**Comment by cclerget** [comment]
@mem Well when I add tests like in #4584 and have to wait 1 minute each time I run a test alone or the array of tests by waiting than `e2e.EnsureImage` finished to build the test image ... it's annoying to wait 1 minute for a test execution of 1 second and see the test failed and so on ... So basically this change would allow when adding tests to do :

```
singularity build -f /tmp/test_image.sif e2e/testdata/Singularity
SINGULARITY_E2E_IMAGE=/tmp/test_image.sif ./scripts/e2e-test -v -run TestE2E/ACTIONS/binds ./e2e
```

The test failed ... fix it and re-run without losing 1 minute again :

**Review by mem — CHANGES_REQUESTED** [review]
Given your previous comment, I understand why you want this sort of optimization, but you are introducing unnecessary variability into the tests. This sort of change does not generalize well: now there would be a branch where there should be none; it's changing the behavior of the tests based on a value taken from the environment; the image is not controlled by the tests (its contents is unknown).

This is the kind of thing that you keep in a local patch, but not one to be pushed to the repository. I'm sorry I cannot provide a better answer, but the nature of the problem you want to solve does not have a better answer, as you have to pick one: either you make the tests deterministic, or you make them flexible.

Taking a step back, the only reason you mentioned previously for this change is the time it takes to run a test. If the problem is that the tests are slow because they need to build this image, maybe the problem that needs fixing is that: identify a _very small_ image that can be used for this purpose and commit _that_ (along with its corresponding definition file) to the repository. We already have a couple of those in the repo, and that's a problem in itself (there's more than one). We are also downloading images from the network for cases where we are _not_ testing the download path but something else. If we can create a single _very small_ image that can be shared by all these tests, we would be fixing an actual problem that's impacting a lot of people.

**Comment by cclerget** [comment]
@mem Changed for alpine resulting in a 10 minutes gain :+1: . I checked tests wasn't skipped and it looks good so far, just I noticed that encryption tests are actually skipped

File: scripts/e2e-test [hunk 1]
```diff
@@ -16,5 +16,6 @@ sudo \
 	env -i \
 		PATH=$PATH \
 		HOME=$HOME \
+		SINGULARITY_E2E_TESTIMAGE=$SINGULARITY_E2E_TESTIMAGE \
```
Comment (dtrudg) [inline scripts/e2e-test]: @cclerget - this line is now redundant I think?

File: e2e/suite.go [hunk 2]
```diff
@@ -116,7 +116,7 @@ func Run(t *testing.T) {
 	}
 	testenv.TestDir = name
 
-	// Build a base image for tests
+	// Build a base image for tests or use the provided one
```
Comment (mem) [inline e2e/suite.go]:
```suggestion
	// Build a base image for tests
```

File: scripts/e2e-test (hunk 1 above)
Comment (mem) [inline scripts/e2e-test]:
```suggestion
```

**Review by mem — APPROVED** [review]
thanks! that's good news :-)

**Review by mem — CHANGES_REQUESTED** [review]
Just commit those changes

**Comment by jscook2345** [comment]
Alpine is kinda of a specialist container OS, without glibc yes? Are there other options for a small image but a more standard OS? I feel like our users do not use alpine often.

update: This is more of a general question. For this issue I think its fine to use alpine.

**Comment by mem** [comment]
> Alpine is kinda of a specialist container OS, without glibc yes? Are there other options for a small image but a more standard OS? I feel like our users do not use alpine often.

Yeah, but the point is that we need an image to test certain singularity commands. What we are testing here is that singularity does what we expect it to do for a given command (IIRC there's a test that does something like `singularity exec test.sif true`), not that CentOS provides enough jun^Wfeatures by default to make a test pass ;-)

**Review by mem — APPROVED** [review]
Thanks!

**Comment by mem** [comment]
> > Alpine is kinda of a specialist container OS, without glibc yes? Are there other options for a small image but a more standard OS? I feel like our users do not use alpine often.
> 
> Yeah, but the point is that we need an image to test certain singularity commands. What we are testing here is that singularity does what we expect it to do for a given command (IIRC there's a test that does something like `singularity exec test.sif true`), not that CentOS provides enough jun^Wfeatures by default to make a test pass ;-)

And just to expand on that thought:

We need a series of tests that test basic singularity functionality (like the above) and then we need a series of tests that check that singularity works on a given distribution (Ubuntu, CentOS, etc) and then we need some tests that test that specific containers (or containers based on specific distros) work. While it might look like we are doing the later with the image as it is before this change, I really don't think that's the case.

Does that make sense?

**Pull request merged**